src/sentry/tasks/data_export.py

```diff
@@ -13,27 +13,38 @@
 SNUBA_MAX_RESULTS = 1000
 
 
+class DataExportError(Exception):
+    pass
+
+
 @instrumented_task(name="sentry.tasks.data_export.assemble_download", queue="data_export")
 def assemble_download(data_export):
     # Create a temporary file
-    with tempfile.TemporaryFile() as tf:
-        # Process the query based on its type
-        if data_export.query_type == ExportQueryType.DISCOVER_V2:
-            process_discover_v2(data_export, tf)
-            return
-        elif data_export.query_type == ExportQueryType.BILLING_REPORT:
-            process_billing_report(data_export, tf)
-            return
-        elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
-            file_name = process_issue_by_tag(data_export, tf)
-        # Create a new File object and attach it to the ExportedData
-        tf.seek(0)
-        with transaction.atomic():
-            file = File.objects.create(
-                name=file_name, type="export.csv", headers={"Content-Type": "text/csv"}
-            )
-            file.putfile(tf)
-            data_export.finalize_upload(file=file)
+    try:
+        with tempfile.TemporaryFile() as tf:
+            # Process the query based on its type
+            if data_export.query_type == ExportQueryType.DISCOVER_V2:
+                process_discover_v2(data_export, tf)
+                return
+            elif data_export.query_type == ExportQueryType.BILLING_REPORT:
+                process_billing_report(data_export, tf)
+                return
+            elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
+                file_name = process_issue_by_tag(data_export, tf)
+            # Create a new File object and attach it to the ExportedData
+            tf.seek(0)
+            try:
+                with transaction.atomic():
+                    file = File.objects.create(
+                        name=file_name, type="export.csv", headers={"Content-Type": "text/csv"}
+                    )
+                    file.putfile(tf)
+                    data_export.finalize_upload(file=file)
+            except IntegrityError:
+                raise DataExportError("Failed to save the assembled file")
+    except DataExportError as err:
+        # TODO(Leander): Implement logging
+        return data_export.email_failure(message=err)
```

Comment:
@leeandher I would think that you'd want information about what the user did such as the `query_type` or even the query that failed if that's readable.

Reply:
That makes sense -> 
<img width="1054" alt="image" src="https://user-images.githubusercontent.com/35509934/75397691-59a7a680-58ac-11ea-83e6-2fda24b663d3.png">

Reply:
@leeandher that looks great!